test/types/index.ts

```diff
@@ -8,6 +8,12 @@ agent.start({
 function started (bool: boolean) {}
 started(agent.isStarted())
 
+agent.setFramework({})
+agent.setFramework({ name: 'foo' })
+agent.setFramework({ name: 'foo', version: 'bar' })
+agent.setFramework({ version: 'bar' })
```

Comment:
Should we consider it valid to include `version` yet omit `name`? Not sure how useful it is to have a version for something when we don't know what it is. 😅

Reply:
Maybe the name was set via the `frameworkName` config option? I agree its probably not that common, but I didn't see a reason to disallow it.

Reply:
Ah, that's fair.